### ISSUE TITLE
Cool, but.... field names.

### ISSUE THREAD

**Issue by myearwood1** [ChristianEhlscheid/vfp2c32#17]
Recursion works nicely. FullPath is an excellent addition. Good decision to make it a memo field. 

Fieldnames is also excellent. I stubbed my toe on it as I did not expect them to be required to be lower case. This failed: Filename,writetime,filesize,fileattribs. Please accept them in any case, and make them lower case internally. Although, I can handle that in DIRX.prg. I'll never call upon adirex directly, so I'll always be free to adapt to your changes, and testing any new implementation will always be a simple thing:

`adirex('c_mycurs','c:\somefolder\reports\*.*',GetADirExAttribs('RASHDF'),GetADirExDestType('CURSOR')+GetADirExVarious('RECURSE')+GetADirExVarious('FULLPATH'),0,"filename,writetime,filesize,fileattribs")
`
GetADirExVarious.prg
```
LPARAMETERS m.tcFunctionType
#include vfp2c.h
LOCAL lcFunctionType
lcFunctionType=iif(vartype(m.tcFunctionType)<>"C","",UPPER(m.tcFunctionType))
DO CASE
    CASE m.lcFunctionType = "RECURSE"
        RETURN ADIREX_RECURSE
    CASE m.lcFunctionType = "FULLPATH"
        RETURN ADIREX_FULLPATH
    CASE m.lcFunctionType = "DISABLE_FSREDIRECTION"
        RETURN ADIREX_DISABLE_FSREDIRECTION
    OTHERWISE
        RETURN -1  && Invalid filter type
ENDCASE

```

**Comment by ChristianEhlscheid** [comment]
Hello,

I've already made the following changes:
- the third parameter to AdirEx can now also be passed as a string - a 1 to 1 translation of your GetADirExAttribs function
- the fieldnames comparison is done case insensitive
- in the process I also discovered a bug in the errorhandling code - this will affect backwards compatibility

I need a few days to update the documentation and make a new release, if you have more suggestions to improve the code feel free to do so.

Regards
Christian

**Comment by myearwood1** [comment]
That is well done and I will try it soon. Thanks

My DirX routine will also separate out the Attributes as logical values. 

It may be that one will want to convert RASHD <-> nAttributes, so I also created GetADirExCAttribs.PRG

```
LPARAMETERS m.tnAttributes

#INCLUDE vfp2c.h

LOCAL m.lcAttributes, m.lnAttributes, m.lnPos
m.lnAttributes = IIF(inlist(VARTYPE(m.tnAttributes),"N","I"), m.tnAttributes, 0)

IF m.lnAttributes = 0
	return ''
endif

DIMENSION laAttribs[7,2]
laAttribs[1,1]=FILE_ATTRIBUTE_READONLY
laAttribs[1,2]="R"
laAttribs[2,1]=FILE_ATTRIBUTE_ARCHIVE
laAttribs[2,2]="A"
laAttribs[3,1]=FILE_ATTRIBUTE_SYSTEM
laAttribs[3,2]="S"
laAttribs[4,1]=FILE_ATTRIBUTE_HIDDEN
laAttribs[4,2]="H"
laAttribs[5,1]=FILE_ATTRIBUTE_DIRECTORY
laAttribs[5,2]="D"
laAttribs[6,1]=FILE_ATTRIBUTE_FAKEDIRECTORY
laAttribs[6,2]="F"
laAttribs[7,1]=FILE_ATTRIBUTE_COMPRESSED
laAttribs[7,2]="C"

lcAttributes=''
FOR m.lnPos = 1 TO ALEN(laAttribs,1)
	IF BITAND(m.lnAttributes,laAttribs[m.lnPos,1])=laAttribs[m.lnPos,1]
	    m.lcAttributes = m.lcAttributes + laAttribs[m.lnPos,2]
	ENDIF
ENDFOR

RETURN m.lcAttributes

```

**Comment by ChristianEhlscheid** [comment]
Hello Mike,

yes you're right, I will fix that before the next release.
I was almost finished to release a new version today, this will postpone it by a few hours ...

**Comment by ChristianEhlscheid** [comment]
Hello Mike,

this was quicker to fix than I thought, I just released the new version.

**Comment by myearwood1** [comment]
THIS IS CRAZY FAST!

I'm trying to make dirX.prg provide the features of DOS DIR. 

I selected just the filename, writetime, filesize and cfileattribs. I populated the folder (since Fullpath works all the time) and fileext. cFileAttribs automatically populated without needing the ADIREX_FILEATTRIBUTES_STRING. That is very intuitive!

I attempted to get a list of files with the A attribute and Exact match and recursion. That's because I'm trying to make code changes in my dev folder, then push them to a test folder before checking in to source control. It works well.
